kernel/common/hash.rb

```diff
@@ -632,6 +632,9 @@ def [](key)
     end
   end
 
+  # Used internally to get around subclasses redefining #[]
+  alias_method :__get__, :[]
+
```

Comment:
Assuming I went the latter, where should I make such statement? A comment on `to_proc` perhaps?